### PR TITLE
Add basic Entity methods and tests for maintanability

### PR DESCRIPTION
**Pull request by zizouet**
I added some basic methods to all entities including :
- [x] Getter
- [x] Equals
- [x] toString
- [x] hashCode
- [x] Private Attributes
- [x] Entity tests
- [x] Add Junit annotations to tests classes to allow test them all at once with tasks.

I also put the main attributes of entities in private to force going by the getters for better maintanability and reduce developer mistakes.

### CODE REVIEW

File: app/src/main/java/com/github/se/assocify/model/entities/User.kt [hunk 1]
```diff
@@ -1,5 +1,30 @@
 package com.github.se.assocify.model.entities
 
-data class User(val uid: String, val name: String, val role: Role) {
+data class User(val uid: String, private val name: String, private val role: Role) {
   constructor() : this("", "", Role("pending"))
+
+  override fun equals(other: Any?): Boolean {
```
Comment (Polymeth) [inline app/src/main/java/com/github/se/assocify/model/entities/User.kt:6]: Not sure why 'other' is of type 'Any?'. In which cases do we compare a role to some other object ?

**Comment by Polymeth** [comment]
In future we would need to formalize having JavaDoc everywhere, but as we have no JavaDoc anywhere I'm not gonna block the PR for this. TODO for later though. Left one question, looks good otherwise it makes the code better :)

File: app/src/main/java/com/github/se/assocify/model/entities/User.kt (hunk 1 above)
Comment (zizouet) [reply app/src/main/java/com/github/se/assocify/model/entities/User.kt:6]: equals is an overriden method which only takes other: Any ? as an argument

**Review by Polymeth — APPROVED** [review]
Alright good work ;)

**Pull request merged**